### DISCUSSION TITLE
MSIX deployment

### DISCUSSION THREAD

**Discussion by renrutsirhc** [AvaloniaUI/Avalonia#6107] [Q&A] [answered]
Has anyone had any luck deploying an Avalonia app using MSIX and the Windows Application Packaging Project? The installer itself is all working fine - the problem is that the app won't launch once installed - it just doesn't do anything. (there's no issues running the unpackaged version)

Anyone come across a similar problem and can suggest anything?

**Comment by maxkatz6** [comment, marked as answer]
There nothing special about Avalonia when using MSIX. 
Log errors in your application. Check windows event viewer for some additional information. 
Also remember that your application does not have write access to the installation folder, because it's shared between users. You need to use AppData or ProgramData folders for mutable data.

**Reply by renrutsirhc** [reply]
Thanks for the hints Max, the event viewer proved the most useful in this case. It turned out that there was nothing wrong with my app at all. The problem is Visual Studio 16.10.0 through 16.10.2 - there's a couple of bugs with packaging msix files when using those versions that mean they don't run correctly. https://developercommunity.visualstudio.com/t/visual-studio-upgrade-to-1610-break-package-compil/1435154
Fortunately, I had another machine around with 16.9.4 on it and the packaging worked a charm first time on it. Hopefully this post will save someone else the hours of wondering what on earth was going on!

**Reply by MelbourneDeveloper** [reply]
@renrutsirhc @maxkatz6 are either of you able to point me to an example of an Avalonia app that is packaged as an MSIX?

**Reply by maxkatz6** [reply]
@MelbourneDeveloper I package my apps with MSIX, but can't share them here.
But this one was successfully packages and published to the store - https://github.com/AvaloniaUI/XamlControlsGallery/tree/master/WindowsStorePackage (though it wasn't updated in ages neither in repo nor in the store).


IIRC, there is no any Avalonia specific issue when you create MSIX package, flow should be the same as with any other .NET 5 application. But feel free to ping me, if you need any help with it.

**Reply by MelbourneDeveloper** [reply]
@maxkatz6 thanks. There are a couple of project files there but the solution doesn't reference them. I tried adding them to the solution but got errors. If anyone has time to dust this off, it would be much appreciated.

![image](https://user-images.githubusercontent.com/16697547/141661761-350ccbe8-9238-479a-b4a6-ed6dde03d99f.png)

**Reply by maxkatz6** [reply]
@MelbourneDeveloper I created PR that includes this project back to the solution https://github.com/AvaloniaUI/XamlControlsGallery/pull/15
Created msix package locally, seems to work without any issues for me.

That repository requires more maintenance and isn't completed. We need more hands :)
